lib/typescript-extensions-requiring-type-checking.js

```diff
@@ -1,23 +1,23 @@
 module.exports = {
     rules: {
         // Defined by Airbnb
-        // 'dot-notation': 'off',
-        // '@typescript-eslint/dot-notation': ['error', { allowKeywords: true }],
+        'dot-notation': 'off',
+        '@typescript-eslint/dot-notation': ['error', { allowKeywords: true }],
 
         // Defined by Airbnb
-        // 'no-implied-eval': 'off',
-        // '@typescript-eslint/no-implied-eval': 'error',
+        'no-implied-eval': 'off',
+        '@typescript-eslint/no-implied-eval': 'error',
```

Comment:
I got two errors from [no-implied-eval](https://github.com/typescript-eslint/typescript-eslint/blob/498f397ff3898dde631f37311615b555f38a414e/packages/eslint-plugin/docs/rules/no-implied-eval.md) that seem to me to be mistaken. In both cases the code is defining a function then calling `bind` on it before passing it to `setTimeout`. The rule is supposed to trigger if you pass a string to `setTimeout` but it looks like `bind` produces a function, not a string. Does anyone know why this should be an error?
```ts
setTimeout(function () {
    this.modalFilters.show();
}.bind(this), 0);
```

Reply:
Not sure why the bind call would trigger no-implied-eval. Their examples even include something pretty similar as a correct usage example:
```ts
setTimeout(foo.fn.bind(this), 100);
```

But this case is probably better written with an arrow function to bind the this context as:
```ts
setTimeout(() => this.modalFilters.show(), 0);
```

Reply:
General agreement that we should keep this rule. We have other rules that should enforce the arrow syntax preference. Jesse can file a bug to typescript-eslnt about the bind behavior.

Reply:
I tried to create a reproducible case to file a bug but didn't get anywhere.
- wrapping the function in parens `(function () {...})` still fails in Web Portal
- copying the failing code into the typescript-eslint test suite passes
- using the arrow syntax also passes

Since we have a workaround I don't think this is high enough priority to investigate further.

Reply:
@jattasNI should we report a bug to typescript-eslint? The error does seem incorrect.

Marking the thread as resolved since we will not modify the current config due to the incorrect behavior.